game-core/src/main/java/games/strategy/engine/lobby/client/login/LobbyServerPropertiesFetcher.java

```diff
@@ -77,23 +67,27 @@ public LobbyServerPropertiesFetcher() {
     if (fromHostedFile.isPresent()) {
       return fromHostedFile;
     }
-
     return getLastUsedProperties();
   }
 
   private static Optional<LobbyServerProperties> getTestOverrideProperties() {
-    return getTestOverrideProperties(ClientSetting.testLobbyHost, ClientSetting.testLobbyPort);
+    return getTestOverrideProperties(
+        ClientSetting.testLobbyHost,
+        ClientSetting.testLobbyPort,
+        ClientSetting.httpLobbyUriOverride);
   }
 
   @VisibleForTesting
   static Optional<LobbyServerProperties> getTestOverrideProperties(
       final GameSetting<String> testLobbyHostSetting,
-      final GameSetting<Integer> testLobbyPortSetting) {
-    if (testLobbyHostSetting.isSet() && testLobbyPortSetting.isSet()) {
+      final GameSetting<Integer> testLobbyPortSetting,
+      final GameSetting<String> testLobbyHttpUri) {
+    if (testLobbyHostSetting.isSet() && testLobbyPortSetting.isSet() && testLobbyHttpUri.isSet()) {
       return Optional.of(
           LobbyServerProperties.builder()
               .host(testLobbyHostSetting.getValueOrThrow())
               .port(testLobbyPortSetting.getValueOrThrow())
+              .httpServerUri(URI.create(testLobbyHttpUri.getValueOrThrow()))
```

Comment:
This and line 108 below are almost screaming for a type-safe `GameSetting<URI>` in the future. :smile: